ADR/0027-tekton-results-naming-convention.md

```diff
@@ -0,0 +1,150 @@
+are evaluating. [[ADR-0014](0014-let-pipelines-proceed.html)]
+
+If tasks exist outside of the build pipeline, they _may_ adhere to the following decisions or they may fall back to the status (failing or succeeding) of a TaskRun.
+
+There are two defined standards for minimized [Tekton result](https://tekton.dev/docs/pipelines/tasks/#emitting-results) formats based on the common task types -- test-like and scan-like tasks. Each of these standards will have a unique result name as well as their own result format.
+
+The standards presented in this ADR supersede those in [ADR 13. AppStudio Test Stream - API contracts](0013-integration-service-api-contracts.html). All other standards presented in the previous ADR hold unless _this_ ADR is superseded by an additional ADR that deprecates those standards. These other non-deprecated standards presented include [Detailed Conftest Output JSON](0013-integration-service-api-contracts.html#detailed-conftest-output-json), [Information injection](0013-integration-service-api-contracts.html#information-injection), [Information format](0013-integration-service-api-contracts.html#information-format), and [Image references](0013-integration-service-api-contracts.html#image-references).
+
+### Results for Test-like Tasks
+Test-like tasks are those whose results can be immediately classified as successful or failed. If these tests are taken into account in the final suitability of an artifact for promotion or release, then all pass/fail determination would be deferred to the task run.
+
+The output of each Tekton task will be provided in a minimized [Tekton result](https://tekton.dev/docs/pipelines/tasks/#emitting-results) in JSON format listing all test failures. The name of the result will be **TEST_OUTPUT**.
+
+#### Tekton Result Format for `TEST_OUTPUT`
+
+The Test output of the Tekton result **TEST_OUTPUT** will be a JSON object that includes the context about the test along with the list of check names for all failed checks
+
+The output will provide the following information about the overall test result:
+- **result** - The outcome of the testing task, can be `SUCCESS`, `FAILURE`, `WARNING`, `SKIPPED` or `ERROR`
+- **namespace** - The rego namespace for the test policy. If not set, it will be assumed to have the value `default`
+- **timestamp** - An UNIX epoch timestamp of the test completion time
+- **successes** - The number of successful checks in the form of an integer
+- **note** - A short note provided by test workstream to provide additional information about the test
+- **failures** - The number of failed checks in the form of an integer
+- **warnings** - The number of warning checks in the form of an integer
+
+
+#### `TEST_OUTPUT` Schema Validation
+The output of the Tekton result **TEST_OUTPUT** can be validated using the following schema:
+
+```
+{
+  "$schema": "http://json-schema.org/draft/2020-12/schema#",
+  "type": "object",
+  "properties": {
```

Comment:
I think it is a great idea to track the digests that are processed, but I think it would be better as a separate result. This would prevent the bleed-through of implementations around multi-arch/Image Indexes into generic test frameworks. Instead, if a test/scan framework _is_ aware of these situations, it enables us to continue to refine the specification to limit the blast radius.

Some potential result names could be `REFS_PROCESSED`, `IMAGES_PROCESSED`, ~`IMAGE_ASSOCIATIONS`~, `DIGEST_OUTPUT`, `ASSOCIATED_REFS_OUTPUT`, ....

Reply:
This has been addressed in the most recent commit using a result named `IMAGES_PROCESSED`.